src/foam/nanos/ticket/Ticket.js

```diff
@@ -257,6 +262,64 @@ foam.CLASS({
     }
   ],
 
+  methods: [
+    {
+      name: 'authorizeOnCreate',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaThrows: ['AuthorizationException'],
+      javaCode: `
+        // Everyone can create a ticket
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaThrows: ['AuthorizationException'],
+      javaCode: `
+        Subject subject = (Subject) x.get("subject");
+        User user = subject.getRealUser();
+
+        if ( user.getId() != this.getCreatedBy() ) {
+          throw new AuthorizationException("You don't have permission to read this ticket.");
+        }
+      `
+    },
+    {
+      name: 'authorizeOnUpdate',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'oldObj', type: 'foam.core.FObject' }
+      ],
+      javaThrows: ['AuthorizationException'],
+      javaCode: `
+        Subject subject = (Subject) x.get("subject");
+        User user = subject.getRealUser();
+
+        if ( user.getId() != this.getCreatedBy() ) {
+          throw new AuthorizationException("You don't have permission to update this ticket.");
+        }
+      `
+    },
+    {
+      name: 'authorizeOnDelete',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaThrows: ['AuthorizationException'],
+      javaCode: `
+        AuthService auth = (AuthService) x.get("auth");
+        
+        if ( ! auth.check(x, "ticket.delete") )  {
+          throw new AuthorizationException("Yon don't have permission to delete this ticket");
+        }
```

Comment:
can you test if the checkGlobalRemove already account for this?

Reply:
Yes. `checkGlobalRemove` in `StandardAuthorizer` should hit when the admin deletes a ticket.